checkpoints-relayer/src/main.rs

```diff
@@ -0,0 +1,236 @@
+
+            return;
+        }
+    };
+
+    let sync_update = match receiver.recv().await {
+        Some(finality_update) => finality_update,
+        None => {
+            log::info!("Updates receiver has been closed before the loop. Exiting");
+
+            return;
+        }
+    };
+
+    let mut slot_last = sync_update.finalized_header.slot;
+
+    match sync_update::try_to_apply(&client, &mut listener, program_id, sync_update.clone()).await {
+        Status::Ok | Status::NotActual => (),
+        Status::Error => return,
+        Status::ReplayBackRequired {
+            replayed_slot,
+            checkpoint,
+        } => {
+            replay_back::execute(
+                &client_http,
+                &beacon_endpoint,
+                &client,
+                &mut listener,
+                program_id,
+                replayed_slot,
+                checkpoint,
+                sync_update,
+            )
+            .await;
+            log::info!("Exiting");
+            return;
+        }
+    }
+
+    loop {
+        let future_interrupt = signal_interrupt.recv();
+        pin_mut!(future_interrupt);
+
+        let future_update = receiver.recv();
+        pin_mut!(future_update);
+
+        let sync_update = match future::select(future_interrupt, future_update).await {
+            Either::Left((_interrupted, _)) => {
+                log::info!("Caught SIGINT. Exiting");
+                return;
+            }
+
+            Either::Right((Some(sync_update), _)) => sync_update,
+            Either::Right((None, _)) => {
+                log::info!("Updates receiver has been closed. Exiting");
+                return;
+            }
+        };
+
+        let committee_update = sync_update.sync_committee_next_pub_keys.is_some();
+        let slot = sync_update.finalized_header.slot;
+        if slot == slot_last {
+            let metric_message = MetricMessage {
+                slot,
+                committee_update,
+                processed: false,
+            };
+
+            if sender_metrics.send(metric_message).await.is_err() {
+                log::error!("Failed to update metrics. Exiting");
+                return;
+            }
+
+            continue;
+        }
+
+        match sync_update::try_to_apply(&client, &mut listener, program_id, sync_update).await {
+            Status::Ok => {
+                slot_last = slot;
+
+                let metric_message = MetricMessage {
+                    slot,
+                    committee_update,
+                    processed: true,
+                };
+
+                if sender_metrics.send(metric_message).await.is_err() {
+                    log::error!("Failed to update metrics. Exiting");
+                    return;
+                }
+            }
+            Status::NotActual => (),
+            Status::ReplayBackRequired { .. } => {
+                log::info!("Exiting");
+                return;
+            }
+            _ => continue,
```

Comment:
Why you ignore `Error` variant?

Reply:
it maybe LowVoteCount that is not critical and maybe skipped as anothers